### PR TITLE
x86_64: implement more operations

### PR DESCRIPTION
**Pull request by jacobly0**
* Implement on integers:
     - `@max`
     - `@clz`
     - `@ctz`
     - `@popCount`
     - `@byteSwap`
     - `@bitReverse`
     - `@atomicLoad`
     - `@atomicStore`
     - `@atomicRmw`
     - `@cmpxchgWeak`
     - `@cmpxchgStrong`
     - `@fence`
 * Implement on floats:
     - `+`
     - `-`
     - `*`
     - `/`
     - `@divTrunc`
     - `@divFloor`
     - `@min`
     - `@max`
 * Implement:
     - error union stuff
     - optional stuff
     - slice stuff
     - `@frameAddress`
     - `@returnAddress`
 * Improve:
     - error union stuff
     - optional stuff
     - slice stuff
     - `~`
     - `@min`
     - `@memcpy`
     - `@memset`
 * Passing 73% (1144/1575) behavior tests vs llvm

### CODE REVIEW

File: src/arch/x86_64/CodeGen.zig [hunk 1]
```diff
@@ -3580,6 +3545,58 @@ fn genBinOp(
...
         .xor => try self.genBinOpMir(.xor, lhs_ty, dst_mcv, src_mcv),
 
+        .min,
+        .max,
+        => {
+            if (!lhs_ty.isAbiInt() or !rhs_ty.isAbiInt()) {
+                return self.fail("TODO implement genBinOp for {s} {}", .{ @tagName(tag), lhs_ty.fmt(self.bin_file.options.module.?) });
+            }
+
+            const mat_src_mcv = switch (src_mcv) {
```
Comment (kubkon) [inline src/arch/x86_64/CodeGen.zig:3555]: What's `mat_` meant to signify?

Comment (jacobly0) [reply src/arch/x86_64/CodeGen.zig:3555]: materialize

Comment (kubkon) [reply src/arch/x86_64/CodeGen.zig]: Hah, wouldn't have guessed that!

**Comment by kubkon** [comment]
I have a small favour to ask since you are doing such a splendid job with the encoder. Would you mind downstreaming any changes to the encoder to https://github.com/kubkon/zig-dis-x86_64? I envision both merging into one at some point but I don't think we are there yet, and it will be very useful for the x86_64 disassembler and `zig-objdump`. Feel free to decline and I will be happy to do just that myself.

Great work so far by the way!

**Review by kubkon — APPROVED** [review]

File: src/arch/x86_64/Encoding.zig [hunk 2]
```diff
@@ -316,39 +307,63 @@ pub const Mnemonic = enum {
...
+    maxss, minss,
     movss,
+    mulss,
+    subss,
     ucomiss,
     // SSE2
     addsd,
-    cmpsd,
-    movq, movsd,
+    //cmpsd,
```
[the remark this answers is not in the document]
Comment (kubkon) [reply src/arch/x86_64/Encoding.zig:357]: Oh, I think I see now. They were already defined elsewhere alphabetically.

[the remark this answers is not in the document]
Comment (kubkon) [reply src/arch/x86_64/Encoding.zig:357]: So basically we have two colliding mnemonics: `cmpsd` for string comparison in legacy mode, and SSE instruction `cmpsd`. Wow, thanks Intel.

**Pull request merged**